--- quake2-0.3/src/q_sh.c.orig	2003-02-14 05:01:37.000000000 +0100
+++ quake2-0.3/src/q_sh.c	2007-04-28 13:25:17.409546491 +0200
@@ -21,6 +21,11 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
 
+/* for mremap() */
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <errno.h>
--- quake2-0.3.1/src/gl_fxmesa.c~	2010-01-31 21:30:28.000000000 +0200
+++ quake2-0.3.1/src/gl_fxmesa.c	2010-01-31 21:31:31.414537554 +0200
@@ -186,7 +186,7 @@
 	
 	ri.Con_Printf(PRINT_ALL, "...setting mode %d:", mode);
 	
-	if( !ri.Vid_GetModeInfo( &width, &height, mode)){
+	if( !ri.Vid_GetModeInfo( (unsigned int *)&width, (unsigned int *)&height, mode)){
 		ri.Con_Printf( PRINT_ALL, " invalid mode\n");
 		return rserr_invalid_mode;
 	}
--- quake2-0.3.1/src/snd_ao.c~	2010-01-31 16:54:57.000000000 +0200
+++ quake2-0.3.1/src/snd_ao.c	2010-01-31 21:32:28.668015776 +0200
@@ -157,7 +157,7 @@
 	/* ao_play returns success, not number of samples successfully output
 	 * unlike alsa or arts, so we can only assume that the whole buffer
 	 * made it out... though this makes updating si->dma->samplepos easy */
-	if(ao_play(device, si->dma->buffer, si->dma->samples * samplesize) == 0){
+	if(ao_play(device, (char *)si->dma->buffer, si->dma->samples * samplesize) == 0){
 		Com_Printf("W: error occurred while playing buffer\n");
 		ao_close(device);
 		ao_shutdown();
