--- quake2-0.3.1/src/rw_x11.c~	2010-01-31 16:54:57.000000000 +0200
+++ quake2-0.3.1/src/rw_x11.c	2010-01-31 21:45:37.338351984 +0200
@@ -62,7 +62,7 @@
 #include <X11/keysym.h>
 #include <X11/extensions/XShm.h>
 #ifdef HAVE_XF86_DGA
-#include <X11/extensions/xf86dga.h>
+#include <X11/extensions/Xxf86dga.h>
 #endif
 #ifdef HAVE_XF86_VIDMODE
 #include <X11/extensions/xf86vmode.h>
--- quake2-0.3.1/src/gl_glx.c~	2010-01-31 16:54:57.000000000 +0200
+++ quake2-0.3.1/src/gl_glx.c	2010-01-31 21:48:39.901166839 +0200
@@ -64,7 +64,7 @@
 #include <X11/cursorfont.h>
 
 #ifdef HAVE_XF86_DGA
-#include <X11/extensions/xf86dga.h>
+#include <X11/extensions/Xxf86dga.h>
 #endif /* HAVE_XF86_DGA */
 #ifdef HAVE_XF86_VIDMODE
 #include <X11/extensions/xf86vmode.h>
